lib/inferno/dsl/fhir_resource_validation.rb

```diff
@@ -295,14 +294,12 @@ def operation_outcome_from_hl7_wrapped_response(response)
 
         # @private
         def operation_outcome_from_validator_response(response, runnable)
-          if response.body.start_with? '{'
-            operation_outcome_from_hl7_wrapped_response(response.body)
-          else
-            runnable.add_message('error', "Validator Response: HTTP #{response.status}\n#{response.body}")
-            raise Inferno::Exceptions::ErrorInValidatorException,
-                  'Validator response was an unexpected format. ' \
-                  'Review Messages tab or validator service logs for more information.'
-          end
+          operation_outcome_from_hl7_wrapped_response(JSON.parse(response.body))
+        rescue JSON::ParserError
+          runnable.add_message('error', "Validator Response: HTTP #{response.status}\n#{response.body}")
```

Comment:
Because this runnable gets persisted to the DB, I think this might be where the change to handle null characters will need to be.